lib/rules/no-unreachable.js

```diff
@@ -24,12 +24,23 @@ function isInitialized(node) {
 }
 
 /**
- * Checks whether or not a given code path segment is unreachable.
- * @param {CodePathSegment} segment A CodePathSegment to check.
- * @returns {boolean} `true` if the segment is unreachable.
+ * Checks all segments in a set and returns true if all are unreachable.
+ * @param {Set<CodePathSegment>} segments The segments to check.
+ * @returns {boolean} True if all segments are unreachable; false otherwise.
  */
-function isUnreachable(segment) {
-    return !segment.reachable;
+function areAllSegmentsUnreachable(segments) {
+
+    if (segments.size === 0) {
+        return false;
+    }
```

Comment:
This is different from the previous version where `.every` would return `true` for empty arrays of current segments, though I'm not sure if that can ever happen.

Reply:
Oops, thanks for pointing that out.